internal/plugin/service/externalidentity/external_identity_data_source.go

```diff
@@ -0,0 +1,141 @@
+package externalidentity
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aiven/aiven-go-client/v2"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/aiven/terraform-provider-aiven/internal/plugin/util"
+)
+
+var (
+	_ datasource.DataSource              = &externalIdentityDataSource{}
+	_ datasource.DataSourceWithConfigure = &externalIdentityDataSource{}
+	_ util.TypeNameable                  = &externalIdentityDataSource{}
+)
+
+func NewExternalIdentityDataSource() datasource.DataSource {
+	return &externalIdentityDataSource{}
+}
+
+type externalIdentityDataSource struct {
+	client   *aiven.Client
+	typeName string
+}
+
+// externalIdentityDataSourceModel is the model for the external_identity data source.
+type externalIdentityDataSourceModel struct {
+	OrganizationID      types.String `tfsdk:"organization_id"`
+	InternalUserID      types.String `tfsdk:"internal_user_id"`
+	ExternalUserID      types.String `tfsdk:"external_user_id"`
+	ExternalServiceName types.String `tfsdk:"external_service_name"`
+}
+
+// Metadata returns the metadata for the external_identity data source.
+func (r *externalIdentityDataSource) Metadata(
+	_ context.Context,
+	req datasource.MetadataRequest,
+	resp *datasource.MetadataResponse,
+) {
+	resp.TypeName = req.ProviderTypeName + "_external_identity"
+	r.typeName = resp.TypeName
+}
+
+// TypeName returns the data source type name for the external_identity data source.
+func (r *externalIdentityDataSource) TypeName() string {
+	return r.typeName
+}
+
+// Schema defines the schema for the external_identity data source.
+func (r *externalIdentityDataSource) Schema(
+	_ context.Context,
+	_ datasource.SchemaRequest,
+	resp *datasource.SchemaResponse,
+) {
+	resp.Schema = schema.Schema{
+		Description: "Map external service user with internal aiven user",
+		Attributes: map[string]schema.Attribute{
+			"organization_id": schema.StringAttribute{
+				Description: "The organization where internal_user_id exists",
+				Required:    true,
+			},
+			"internal_user_id": schema.StringAttribute{
+				Description: "The identity of the user on aiven platform",
+				Required:    true,
+			},
+			"external_user_id": schema.StringAttribute{
+				Description: "The identity of the user on external platform",
+				Required:    true,
+			},
+			"external_service_name": schema.StringAttribute{
+				Description: "The name of the external service where external_user_id exists",
```

Comment:
I've just manually added github as an option here because I couldn't see where to get the possible choices from.
```suggestion
				Description:  userconfig.Desc("The name of the external service.").PossibleValuesString("github").Build()
```